doc-bridgepoint/notes/9932_prj_export_int.md

```diff
@@ -0,0 +1,142 @@
+5.2.1.3  This meant that the in-memory model data was being set up as expected but the changes
+were not being persisted.   
+
+5.3  The `masl_round_trip` tool uses the BridgePoint CLI "Launch" mode which avoids starting up
+and shutting down many instances of eclipse and instead starts a single instance which can 
+perform many actions.  This is the reason `masl_round_trip` was succeeding, because it was running
+the pre-build step on the correct in-memory data.  
+
+### 6. Work Required
+
+6.1  `ImportHelper.resolveMASLProject()`  
+6.1.1  At the beginning of this function, the flow starts a transaction by calling 
+`TransactionManager.startTransactionsOnSelectedModelRoots(...)`.  As the function name suggests, the
+transaction uses model roots that are part of the selection.  Via step debugging, we see that in the
+current flow there are no model elements in the selection, thus, no transaction is created.  The code is
+modified to add the `SystemModel_c` instance to the selection before making the call so that a transaction
+is created.  
+
+6.1.2  At the end of this function there is code that executes outside of the transaction.  After discussion
+among the engineering team, there appears to be no reason for this.  The call to `synchronizeWithMASLDomain(system)`
+is moved up inside the transaction.  
+
+6.2  Change the CLI `ImportExecutor` to call `<SystemModel_c>.persistSelfAndChildren()` after the 
+project or file is imported.  This will force persist the in-memory data before the application exits.  
+
+### 7. Implementation Comments
+
+7.1  During the CLI application shutdown process eclipse may throw exceptions like: 
+```
+org.eclipse.e4.core.di.InjectionException: org.eclipse.swt.SWTException: Invalid thread access
+``` 
+We researched a solution for this issue but turned up nothing.  We also attempted to catch and hide
+the error, but again without success.  The problem is happening in eclipse code that is outside of BridgePoint
+control.  We are calling the proper function `IWorkbench.close()` to exit the application.  No solution
+to making this error go away was found.   
+
+7.2  Removed obsolete and unused function `migrateSLDTs` from `ImportHelper.java`.  The function
+has no implementation and is not called from anywhere.    
+
+7.3  It is not clear why the resolution and assignment of component references inside the (now properly 
+constructed) transaction in `ImportHelper` does not cause a persist of the in-memory data to occur. It could 
+be that there is a race condition between persistence launched by a listener and the application shutdown 
+is happening.  Whatever the cause, it is clear that making sure the CLI Import persists everything before
+shutting down by making a direct invocation is a clear fix for this issue.  
+
+### 8. Unit Test
+
+8.1 Run the full `masl_round_trip` suite and verify success.    
+
+8.2 Run `masl_round_trip` on SAC_OOA and SAC_PROC and verify the `.tr` files are output for SAC_PROC.  
```

Comment:
Please add a short description of how to do this. Also, I got lots of MASL formatter errors in the output, however the .tr files were produced